Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py

```diff
@@ -66,54 +76,85 @@
-    :param unsecure: trust any cert
+    :param unsecure: trust any certificate
     :type unsecure: boolean
+    :param ssl_version: ssl version
+    :type unsecure: string
     :return: ldap3 Server
     :rtype: Server
     """
-
     if secure_connection == "TLS":
+        # Kept the TLS option for backwards compatibility only.
+        # For establishing a secure connection via SSL/TLS protocol - use the 'SSL' option.
+        # For establishing a secure connection via Start TLS - use the 'Start TLS' option.
         demisto.debug(f"initializing sever with TLS (unsecure: {unsecure}). port: {port or 'default(636)'}")
         if unsecure:
             # Add support for all CIPHERS_STRING
-            tls = Tls(validate=ssl.CERT_NONE, ciphers=CIPHERS_STRING)
+            tls = Tls(validate=ssl.CERT_NONE, ciphers=CIPHERS_STRING, version=get_ssl_version(ssl_version))
         else:
-            tls = Tls(validate=ssl.CERT_NONE)
+            tls = Tls(validate=ssl.CERT_NONE, version=get_ssl_version(ssl_version))
         if port:
             return Server(host, port=port, use_ssl=unsecure, tls=tls)
         return Server(host, use_ssl=unsecure, tls=tls)
 
-    if secure_connection == "SSL":
-        # intialize server with ssl
-        # port is configured by default as 389 or as 636 for LDAPS if not specified in configuration
-        demisto.debug(f"initializing sever with SSL (unsecure: {unsecure}). port: {port or 'default(636)'}")
-        # If Trust any certificate is false
-        if not unsecure:
-            demisto.debug("will require server certificate.")
-            tls = Tls(validate=ssl.CERT_REQUIRED, ca_certs_file=os.environ.get('SSL_CERT_FILE'))
-            if port:
-                return Server(host, port=port, use_ssl=True, tls=tls)
-            return Server(host, use_ssl=True, tls=tls)
-        else:
-            # Add support for all CIPHERS_STRING
-            tls = Tls(ciphers=CIPHERS_STRING)
-            if port:
-                return Server(host, port=port, use_ssl=True, tls=tls)
-            return Server(host, use_ssl=True, tls=tls)
-    demisto.debug(f"initializing server without secure connection. port: {port or 'default(389)'}")
-    if port:
-        return Server(host, port=port)
-    return Server(host)
+    if secure_connection == 'SSL':  # Secure connection (SSL\TLS)
+        demisto.info(f"Initializing LDAP sever with SSL/TLS (unsecure: {unsecure})."
+                     f" port: {port or 'default(636)'}")
+        tls = get_tls_object(unsecure, ssl_version)
+        return Server(host=host, port=port, use_ssl=True, tls=tls, connect_timeout=DEFAULT_TIMEOUT)
+
+    elif secure_connection == 'Start TLS':  # Secure connection (STARTTLS)
+        demisto.info(f"Initializing LDAP sever without a secure connection - Start TLS operation will be executed"
+                     f" during bind. (unsecure: {unsecure}). port: {port or 'default(389)'}")
+        tls = get_tls_object(unsecure, ssl_version)
```

Comment:
Because I think that the passed tls object will be used when we are upgrading the security level by executing the 'tls start' command, but let me verify that to be sure.

Reply:
@glicht - according to the start_tls func in site-packages/ldap3/core/connection.py it seems that the passed Tls object is used.